pypdf/_reader.py

```diff
@@ -314,6 +314,7 @@ def __init__(
 
         if isinstance(stream, (str, Path)):
             stream = open(stream, "rb")  #  noqa: SIM115
+            # Wish I could just close stream in __del__ but that fails a test very strangely
```

Comment:
Just out of curiosity: Do you have some details about the failure?

Reply:
Yeah, I'm not sure how much was relevant to drop in the commit but:

when adding a `self.stream.close()` in a `__del__` function, that *does* work most of the time. 

The one test failure I was seeing was in `tests/test_reader.py`, the failing test was `test_get_page_of_encrypted_file` but interestingly this would pass on it's own. I narrowed down the source of the issue to the previous test `test_issue297`'s exception block where the `PdfReader()` initializer was failing (that's what the test is testing for) and the `__del__` block wasn't being called due to the exception happening in the `__init__`. 

It's very possible at some point the objects would be GCd but test failures were happening due to dangling file pointers at the following test.

Reply:
I'm going to add this to the commit